### PR TITLE
ENG-2496 conversion of cms widget config tests to page object pattern

### PR DESCRIPTION
**Pull request by jeffgo10**
+ edit config for cms content widget
+ page object pattern for page designer and widget form
+ removed old commands most especially in page designer
+ reordered test heirarchy in MFE tests
- widget crud not included yet

### CODE REVIEW

File: ui-tests/cypress/support/pageObjects/app/AppPage.js [hunk 1]
```diff
@@ -53,6 +53,12 @@ export default class AppPage extends WebElement {
     return this.content;
   }
 
+  getModalDialogByTitle(title) {
```
Comment (g-di-martino) [inline ui-tests/cypress/support/pageObjects/app/AppPage.js:56]: What is this needed for? Do you have multiple dialog at the same time?
There's already a getDialog method; if it is behaving differently for a specific page you could just override it for that page

File: ui-tests/cypress/support/pageObjects/app/Modal.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+export default class Modal {
```
Comment (g-di-martino) [inline ui-tests/cypress/support/pageObjects/app/Modal.js:1]: Isn't this the same of Dialog class??

File: ui-tests/cypress/support/pageObjects/components/mfeWidgets/MFEWidgetsPage.js [hunk 3]
```diff
@@ -1,5 +1,68 @@
...
+    return this.get()
+      .children(this.maincontent);
+  }
+
+  getListArea() {
+    return this.getContents()
+      .find(this.rowlayout).eq(2);
+  }
+
+  getKebabMenuID(widgetCode) {
```
Comment (g-di-martino) [inline ui-tests/cypress/support/pageObjects/components/mfeWidgets/MFEWidgetsPage.js:28]: Is this needed just for the following method? Why not just massing `widgetCode` to it?!

File: ui-tests/cypress/support/pageObjects/components/mfeWidgets/MFEWidgetsPage.js [hunk 4]
```diff
@@ -1,5 +1,68 @@
...
+  }
+
+  getFooterArea() {
+    return this.getContents()
+      .children(this.rowlayout).eq(3);
+  }
+
+  getAddButton() {
+    return this.getFooterArea()
+               .contains(/^Add$/);
```
Comment (g-di-martino) [inline ui-tests/cypress/support/pageObjects/components/mfeWidgets/MFEWidgetsPage.js:55]: Is `contains` the only way to identify this element?

File: ui-tests/cypress/support/pageObjects/pages/designer/widgetconfigs/ContentQueryWidgetConfigPage.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import WidgetConfigPage from '../WidgetConfigPage';
+
+export default class ContentQueryWidgetConfigPage extends WidgetConfigPage {
+
+  getContentTypeField() {
+    return this.getMainContainer().find('[name="contentType"]');
+  }
+
+  getPublishSettingsAccordButton() {
+    return this.getMainContainer().contains(/^Publishing settings$/i);
```
Comment (g-di-martino) [inline ui-tests/cypress/support/pageObjects/pages/designer/widgetconfigs/ContentQueryWidgetConfigPage.js:10]: Avoid use contains to define the structure of a page

**Review by g-di-martino — CHANGES_REQUESTED** [review]
I know the page designer is completely different from all the other pages, just try to describe it as close as possible to the common page structure

File: ui-tests/cypress/support/pageObjects/components/mfeWidgets/MFEWidgetsPage.js (hunk 3 above)
Comment (jeffgo10) [reply ui-tests/cypress/support/pageObjects/components/mfeWidgets/MFEWidgetsPage.js:28]: this is needed by 2 sibling methods. so I would just stick this method in.

File: ui-tests/cypress/support/pageObjects/app/AppPage.js (hunk 1 above)
Comment (jeffgo10) [reply ui-tests/cypress/support/pageObjects/app/AppPage.js]: highlighted method has been removed. instead used `getDialog` method

File: ui-tests/cypress/support/pageObjects/app/Modal.js (hunk 2 above)
Comment (jeffgo10) [reply ui-tests/cypress/support/pageObjects/app/Modal.js]: Modal class has now been removed

File: ui-tests/cypress/support/pageObjects/components/mfeWidgets/MFEWidgetsPage.js (hunk 4 above)
Comment (jeffgo10) [reply ui-tests/cypress/support/pageObjects/components/mfeWidgets/MFEWidgetsPage.js]: expanded the specificity of the method

File: ui-tests/cypress/support/pageObjects/pages/designer/widgetconfigs/ContentQueryWidgetConfigPage.js (hunk 5 above)
Comment (jeffgo10) [reply ui-tests/cypress/support/pageObjects/pages/designer/widgetconfigs/ContentQueryWidgetConfigPage.js]: specificity of this method has been expanded

File: ui-tests/cypress/support/pageObjects/components/mfeWidgets/MFEWidgetsPage.js (hunk 3 above)
Comment (jeffgo10) [reply ui-tests/cypress/support/pageObjects/components/mfeWidgets/MFEWidgetsPage.js]: shoot, i was looking at a different class. I removed it. should be okay now

File: ui-tests/cypress/support/pageObjects/app/Dialog.js [hunk 6]
```diff
@@ -67,6 +67,11 @@ export default class Dialog extends WebElement {
                .children(htmlElements.button).eq(1);
   }
 
+  getFooterActionByLabel(label) {
+    return this.getFooter()
+      .children(htmlElements.button).contains(label);
```
Comment (g-di-martino) [inline ui-tests/cypress/support/pageObjects/app/Dialog.js:72]: Please, do not use `contains` in page objects.
This class already has methods to close, cancel and confirm the operation by the way

File: ui-tests/cypress/support/pageObjects/pages/designer/widgetconfigs/ContentWidgetConfigPage.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
+import { htmlElements } from '../../../WebElement';
+
+import Dialog from '../../../app/Dialog';
+
+import WidgetConfigPage from '../WidgetConfigPage';
+
+export class ContentListSelectModal extends Dialog {
```
Comment (g-di-martino) [inline ui-tests/cypress/support/pageObjects/pages/designer/widgetconfigs/ContentWidgetConfigPage.js:7]: I've created an additional class to handle different dialog body contents on another [PR](https://github.com/entando/entando-test-automation/pull/14/commits/c46c59d90e5efc375884e5b129d880cabd1f4642); what do you think about it?

**Review by g-di-martino — CHANGES_REQUESTED** [review]

File: ui-tests/cypress/support/pageObjects/pages/designer/widgetconfigs/ContentWidgetConfigPage.js (hunk 7 above)
Comment (jeffgo10) [reply ui-tests/cypress/support/pageObjects/pages/designer/widgetconfigs/ContentWidgetConfigPage.js:7]: looks okay. I'll implement it when that PR goes to merge and I am also working on the next task that still involves on MFE/Widgets so by that time I supposed your PR will get to merge. is that okay?

File: ui-tests/cypress/support/pageObjects/app/Dialog.js (hunk 6 above)
Comment (jeffgo10) [reply ui-tests/cypress/support/pageObjects/app/Dialog.js]: done

File: ui-tests/cypress/support/pageObjects/pages/designer/widgetconfigs/ContentWidgetConfigPage.js (hunk 7 above)
Comment (g-di-martino) [reply ui-tests/cypress/support/pageObjects/pages/designer/widgetconfigs/ContentWidgetConfigPage.js:7]: sure! 😄

**Review by g-di-martino — APPROVED** [review]

**Pull request merged**